clients/src/main/java/io/strimzi/http/producer/HttpProducerClient.java

```diff
@@ -104,11 +113,21 @@ public void checkAndSendMessages() {
     }
 
     public ProducerRecord generateMessage(int numOfMessage) {
-        String record = "{\"records\":[{\"key\":\"key-" + numOfMessage + "\",\"value\":\"" + configuration.getMessage() + "-" + numOfMessage + "\"}]}";
+        String message;
+        if (this.configuration.getMessageTemplate() != null) {
+            message = dataGenerator.generateStringData();
+        } else {
+            message = configuration.getMessage() + "-" + numOfMessage;
+        }
+        if (Objects.equals(this.configuration.getMessageType(), MessageType.text.name())) {
+            message = "\"" + message + "\"";
+        }
+
+        String record = "{\"records\":[{\"key\":\"key-" + numOfMessage + "\",\"value\":" + message + "}]}";
 
         HttpContext context = HttpContext.post(
             configuration.getUri(),
-            "application/vnd.kafka.json.v2+json",
+            "application/vnd.kafka." + this.configuration.getMessageType() + ".v2+json",
```

Comment:
Not sure about this one? One possible realisation is that it would be `application/vnd.kafka.plain.v2+json` or `application/vnd.kafka.Unknown.v2+json` is that correct?

Reply:
it will eventually end-up with some error from the bridge and probably the client will just end with some error code. I think this might be a little bit updated in the config and in case there will be unknown option, it will return IllegallArgument exception or something like that. WDYT?

Reply:
Yeah I would throw an exception in the config class.
+1